backend/sftp/sftp.go

```diff
@@ -1171,6 +1171,11 @@ func (f *Fs) mkdir(ctx context.Context, dirPath string) error {
 	err = c.sftpClient.Mkdir(dirPath)
 	f.putSftpConnection(&c, err)
 	if err != nil {
+		ok, _ := f.dirExists(ctx, dirPath)
```

Comment:
In theory you should be able to use `os.IsExist(err)` and avoid calling `dirExists` here - I don't know how well that works in practice though.

Reply:
I didn't test but `dirExists` suggests that it should - I can test in the actual transfer after current batch finishes though